src/method.c

```diff
@@ -149,6 +152,28 @@ static jl_value_t *resolve_globals(jl_value_t *expr, jl_module_t *module, jl_sve
             if (e->head == method_sym || e->head == module_sym) {
                 i++;
             }
+            if (e->head == new_opaque_closure_sym) {
+                JL_NARGSV(opaque closure definition, 5);
+                for (; i < 4; i++) {
+                    // TODO: this should be making a copy, not mutating the source
+                    jl_exprargset(e, i, resolve_globals(jl_exprarg(e, i), module, sparam_vals, binding_effects, eager_resolve));
+                }
+                jl_value_t *source = jl_exprarg(e, 4);
+                if (jl_is_expr(source) && strcmp(jl_symbol_name(((jl_expr_t*)source)->head), "opaque_closure_method") == 0) {
+                    if (jl_expr_nargs(source) != 3) {
+                        jl_error("opaque_closure_method: invalid syntax");
+                    }
+                    jl_value_t *nargs = jl_exprarg(source, 0);
+                    jl_value_t *functionloc = jl_exprarg(source, 1);
+                    jl_value_t *ci = jl_exprarg(source, 2);
+                    if (!jl_is_code_info(ci)) {
+                        jl_error("opaque_closure_method: lambda should be a CodeInfo");
+                    }
+                    jl_resolve_globals_in_ir((jl_array_t*)((jl_code_info_t*)ci)->code, module, NULL, binding_effects);
```

Comment:
I think this is redundant, since `jl_method_set_source` also calls resolve_globals.

Reply:
Good catch

Reply:
I think, it is a bit more (or less) than that, and `new_opaque_closure_sym` doesn't need any special handling (so, unindent this block), as `opaque_closure_method` handling is independent of its position in the IR?